integration-tests/utils/firecloud-utils.js

```diff
@@ -7,28 +7,39 @@ const selectWorkspace = async (page, billingAccount, workspace) => {
 }
 
 const signIntoFirecloud = async (page, token) => {
+  await page.waitForXPath('//title[text()="FireCloud | Broad Institute"]')
+  await findText(page, 'content you are looking for is currently only accessible')
+
+  await page.waitForXPath('//*[@id="sign-in-button"]')
+
   /*
    * The FireCloud not-signed-in page renders the sign-in button while it is still doing some
    * initialization. If you log the status of the App components state for user-status and auth2
    * with each render, you see the following sequence:
    *   '#{}' ''
    *   '#{}' '[object Object]'
    *   '#{:refresh-token-saved}' '[object Object]'
+   *
    * If the page is used before this is complete (for example window.forceSignedIn adding
    * :signed-in to user-status), bad things happen (for example :signed-in being dropped from
-   * user-status). Instead of reworking the sign-in logic for a case that (for the most part) only
-   * a computer will operate fast enough to encounter, we'll just slow the computer down a little.
+   * user-status). Put another way, if forceSignedIn is called between the 2nd and 3rd render, the
+   * sign-in doesn't work. The visible effect (failure screenshot) is that the test gets stuck on
+   * the FireCloud sign-in page.
+   *
+   * There is no observable evidence that the component has settled into its final state. I (Brian)
```

Comment:
Is settling into its final state based on network requests? Could [`page.waitForNetworkIdle`](https://pptr.dev/#?product=Puppeteer&version=v13.1.3&show=api-pagewaitfornetworkidleoptions) be useful here?

But if this test is passing several hundred runs in a row using a fixed delay, then I'm happy with that approach.